### PR TITLE
Allow "friendsofsymfony/rest-bundle:^3.0"

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

Allow "friendsofsymfony/rest-bundle:^3.0".

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Added
- Added support for "friendsofsymfony/rest-bundle:^3.0"
### Change
- support for deprecated "rest" routing type in favor for xml
```

### CODE REVIEW

File: src/Controller/Api/PostController.php [hunk 1]
```diff
@@ -24,6 +23,7 @@
 use Sonata\NewsBundle\Mailer\MailerInterface;
 use Sonata\NewsBundle\Model\Comment;
 use Sonata\NewsBundle\Model\CommentManagerInterface;
+use Sonata\NewsBundle\Model\Post;
```
Comment (phansys) [inline src/Controller/Api/PostController.php:26]: It seems like a BC break. Is this a bugfix? In that case, it should be done in other PR.

File: src/Resources/config/routing/api/post.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<routes xmlns="http://symfony.com/schema/routing" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/routing https://symfony.com/schema/routing/routing-1.0.xsd">
+    <route id="get_posts" path="/posts.{_format}" methods="GET" controller="Sonata\NewsBundle\Controller\Api\PostController::getPostsAction" format="json">
+        <requirement key="_format">json|xml|html</requirement>
+    </route>
+    <route id="get_post" path="/posts/{id}.{_format}" methods="GET" controller="Sonata\NewsBundle\Controller\Api\PostController::getPostAction" format="json">
```
Comment (phansys) [inline src/Resources/config/routing/api/post.xml:6]: IMO, we could sort these routes by their ids.

File: tests/App/AppKernel.php [hunk 3]
```diff
@@ -95,6 +97,6 @@ protected function configureContainer(ContainerBuilder $containerBuilder, Loader
 
     private function getBaseDir(): string
     {
-        return sys_get_temp_dir().'/sonata-news-bundle/var/';
+        return sys_get_temp_dir().'/sonata-news-bundle/var/8678678678jkhdkdff';
```
Comment (phansys) [inline tests/App/AppKernel.php:100]: What is this change required?

File: tests/Functional/RoutingTest.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\NewsBundle\Tests\Functional\Routing;
+
+use Sonata\NewsBundle\Tests\App\AppKernel;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
```
Comment (phansys) [inline tests/Functional/RoutingTest.php:20]: Thank you for the credit.

File: tests/App/AppKernel.php (hunk 3 above)
Comment (wbloszyk) [reply tests/App/AppKernel.php:100]: It is only fr my local tests.

File: tests/Functional/RoutingTest.php (hunk 4 above)
Comment (wbloszyk) [reply tests/Functional/RoutingTest.php:20]: I copy your work, delete this will be rude.

File: src/Controller/Api/PostController.php (hunk 1 above)
Comment (wbloszyk) [reply src/Controller/Api/PostController.php:26]: Technicly it is BC-break and it is bugfix, anyway it can be change like 'REST' routing type. This will change only docblock, Do we really need this in seperate PR when all RestApi system is rebuilld?

Comment (jordisala1991) [reply src/Controller/Api/PostController.php:26]: IMO we should consider this as a bugfix

Comment (phansys) [reply src/Controller/Api/PostController.php:26]: If this change fixes something what will start working without further modifications, we should create a separate PR.

Comment (wbloszyk) [reply src/Controller/Api/PostController.php:26]: @phansys This is only bug on documentation. It will be working anyway, with or without this change.

@jordisala1991 Do we really need seperate PR for it?

Comment (phansys) [reply src/Controller/Api/PostController.php:26]: >@phansys This is only bug on documentation. It will be working anyway, with or without this change.

Please, note that the documentation is not just intended to be read by humans. It generates definitions compatible with the [Open API Specification](https://swagger.io/docs/specification/2-0/basic-structure/), which is also intended to build API clients (like the sandbox exposed in the HTML documentation).
Wrong specification produces wrong clients.

File: src/Resources/config/routing/api/post.xml (hunk 2 above)
Comment (wbloszyk) [reply src/Resources/config/routing/api/post.xml:6]: I check it. Better keep it in this way becouse order in config is equal to order in `nelmio doc`. Its look much better when we keep this order.

File: src/Controller/Api/PostController.php (hunk 1 above)
Comment (wbloszyk) [reply src/Controller/Api/PostController.php:26]: @phansys Can you do it? You have better knowlage why and how this should be fix. Also will be nice to make [`Update Swagger definition in other to be compatible with string identifiers`](https://github.com/sonata-project/SonataUserBundle/pull/1198) change in other bundles to make RestApi upgrade more complex. 

WDYT?

Comment (phansys) [reply src/Controller/Api/PostController.php]: I'll try, but be patient. I'm very busy these days. I'll try to get some ReST :trollface: this weekend, and in the meantime I'll see if I can provide some PRs.

Comment (wbloszyk) [reply src/Controller/Api/PostController.php]: This PR can be provided independent of my work, so dont worry be happy .

**Comment by phansys** [comment]
Please, check the test results.

**Comment by wbloszyk** [comment]
RTM

File: tests/App/config/config.yaml [hunk 5]
```diff
@@ -4,7 +4,7 @@ imports:
 
 framework:
     test: true
-    secret: secret
+    secret: '50n474.U53r'
```
Comment (phansys) [inline tests/App/config/config.yaml:7]: This was intended to be "SONATA.USER", because this secret was added in SonataUserBundle.

Comment (wbloszyk) [reply tests/App/config/config.yaml:7]: IMHO it will be nice to have this secret in sonata test

**Comment by wbloszyk** [comment]
@jordisala1991 Can you recheck it and ping contributors?

**Comment by VincentLanglet** [comment]
Thanks
